server/webhooks/failing_policies.go

```diff
@@ -128,59 +230,12 @@ type FailingHost struct {
 	URL      string `json:"url"`
 }
 
-func makeFailingHost(host fleet.PolicySetHost, serverURL url.URL) FailingHost {
-	serverURL.Path = path.Join(serverURL.Path, "hosts", strconv.Itoa(int(host.ID)))
+func makeFailingHost(host fleet.PolicySetHost, serverURL *url.URL) FailingHost {
+	u := *serverURL
+	u.Path = path.Join(serverURL.Path, "hosts", strconv.FormatUint(uint64(host.ID), 10))
 	return FailingHost{
 		ID:       host.ID,
 		Hostname: host.Hostname,
-		URL:      serverURL.String(),
-	}
-}
-
-// filterPolicies fetches the policies from the policy set and filters out those
-// that are not configured for webhook anymore or are deleted.
-//
-// The filtered out policies are removed from the set.
-func filterPolicies(
-	ctx context.Context,
-	ds fleet.Datastore,
-	configuredPolicyIDs []uint,
-	failingPoliciesSet fleet.FailingPolicySet,
-	logger kitlog.Logger,
-) ([]*fleet.Policy, error) {
-	configuredPolicyIDsSet := make(map[uint]struct{})
-	for _, policyID := range configuredPolicyIDs {
-		configuredPolicyIDsSet[policyID] = struct{}{}
-	}
-	policySets, err := failingPoliciesSet.ListSets()
-	if err != nil {
-		return nil, ctxerr.Wrap(ctx, err, "listing global policies set")
-	}
-	var policies []*fleet.Policy
-	var gcSet []uint
-	for _, policyID := range policySets {
-		if _, ok := configuredPolicyIDsSet[policyID]; !ok {
-			level.Debug(logger).Log("msg", "skipping policy from set, not in config", "id", policyID)
-			gcSet = append(gcSet, policyID)
-			continue
-		}
-		switch policy, err := ds.Policy(ctx, policyID); {
-		case err == nil:
-			policies = append(policies, policy)
-		case errors.Is(err, sql.ErrNoRows):
-			level.Debug(logger).Log("msg", "skipping policy from set, deleted", "id", policyID)
-			gcSet = append(gcSet, policyID)
-		default:
-			return nil, ctxerr.Wrapf(ctx, err, "failing to load global failing policies set %d", policyID)
-		}
-	}
-	// Remove the policies that are present in the set but:
-	//	- are not present in the config (user disabled automation for them), or,
-	//	- do not exist anymore (user deleted the policy).
```

Comment:
I think the new code is missing one case handled in the deleted code here:

- `are not present in the config (user disabled automation for them), or,` for global policies, that is a similar as the below `if` in `TriggerFailingPoliciesWebhook` but for global policies:

```go
_, ok := teamPolicyIDs[*policy.TeamID][policy.ID]
if !ok {
	level.Debug(logger).Log("msg", "skipping failing policy, not found in team policy IDs", "policyID", policyID)
	if err := failingPoliciesSet.RemoveSet(policy.ID); err != nil {
		level.Error(logger).Log("msg", "failed to remove policy from set", "policyID", policyID, "err", err)
	}
	continue
}
```

Reply:
Good eye, added in f11d13f3d